### DISCUSSION TITLE
Instructions to Host UI in Azure for frontend application giving errors

### DISCUSSION THREAD

**Discussion by biggernaaz** [Azure-Samples/graphrag-accelerator#135] [General]
Hi Everyone, 

Hoping @tushardhadiwal or @jgbradley1 can shed some light here as I have been following PR 128 (Adding deployment script for frontend application) and waiting for some frontend UI guidance with this graphrag accelerator. 

Created the app registration with ID tokens enabled, but I keep running into this error in the "Configuring registry credentials..." step:

--------------------
Creating web app identity...
Web app identity created.
Configuring registry credentials...
ERROR: Cannot find user or service principal in graph database for 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'.
--------------

I am still able to launch the webapp and run queries, but wanted to share this and see if others have run into the same issue following this new guidance. 

Many thanks!

**Comment by tushardhadiwal** [comment]
@biggernaaz Can you confirm if you have Owner level permissions on the subscription to which these resources are being deployed? Some of the steps in the script run role assignment commands and that requires those permissions. You may see this error if one of the commands prior to role assignments failed , can you run the script with `set -eux` enabled on top and check if https://github.com/Azure-Samples/graphrag-accelerator/blob/8118923232c74ef6e5210e8609f71559f34c9ff5/frontend/deploy.sh#L103 `WEBAPP_IDENTITY_CLIENT_ID` is getting assigned or not.

**Reply by biggernaaz** [reply]
Hi @tushardhadiwal thank you very much for your quick response! I confirmed that I am Owner on the subscription and also redeployed the frontend to multiple new resource groups with the `set -eux` enabled to test. It still failed with the same error above. 

+ `WEBAPP_IDENTITY_CLIENT_ID`=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx is getting assigned and matches the client ID of the Managed Identity that is created in Azure by the frontend deployment script. 

When I redeploy to the same resource group, it seems to go a bit further and steps into the `configure_app_settings` function and stops here:

----------
+ az webapp config appsettings set --name rgxxxxxxxxxx-playground --resource-group rgxxxxxxxxxx --settings '#' MS GraphRAG FrontEnd environment variables= '#' DEPLOYMENT_URL=https://apim-xxxxxxxxxx.azure-api.net '#' APIM_SUBSCRIPTION_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: not enough values to unpack (expected 2, got 1)


----------
I notice the backend infrastructure code has been updated since 7/25/24 when I deployed the solution accelerator. Do I need to update my backend infrastructure in any way to make the frontend fully operational in an Azure webapp?

I also tried running that recommended `az ad sp create --id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx` and it threw back a warning "When using this permission, the backing application of the service principal being created must in the local tenant" 

Configuring registry credentials...
++ az acr show --name rgxxxxxxxreg --resource-group rgxxxxxx --query id --output tsv
+ ACR_ID=/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/rgxxxxx/providers/Microsoft.ContainerRegistry/registries/xxxxxxxxreg
+ az role assignment create --assignee xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx --role AcrPull --scope /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/rgxxxxxxxxxx/providers/Microsoft.ContainerRegistry/registries/rgxxxxxxxxxx
ERROR: Cannot find user or service principal in graph database for 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'.

**Reply by biggernaaz** [reply]
Adding two more details in case it may be relevant to troubleshooting this. 

1. I am using a sponsored subscription and the azure account I was using to deploy resources is both a Service Administrator and Owner on the subscription
2. I am using Github Codespaces as my deployment env. 

Curious if anyone else has recently deployed the frontend and encountered any trouble. 

Thank you very much!
